### PR TITLE
[bluray] Update libbluray to 0.8.1 and some enhancements

### PR DESCRIPTION
**Pull request by ace20022**

### CODE REVIEW

File: xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp [hunk 1]
```diff
@@ -1128,4 +1128,39 @@ bool CDVDInputStreamBluray::HasMenu()
   return m_navmode;
 }
 
+void CDVDInputStreamBluray::setupPlayerSettings()
```
Comment (xhaggi) [inline xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamBluray.cpp]: First letter in upper case?

File: xbmc/settings/Settings.cpp [hunk 2]
```diff
@@ -196,6 +196,7 @@ const std::string CSettings::SETTING_DVDS_AUTORUN = "dvds.autorun";
 const std::string CSettings::SETTING_DVDS_PLAYERREGION = "dvds.playerregion";
 const std::string CSettings::SETTING_DVDS_AUTOMENU = "dvds.automenu";
 const std::string CSettings::SETTING_DISC_PLAYBACK = "disc.playback";
+const std::string CSettings::SETTING_BLURAY_PLAYERREGION = "bluray.region";
```
Comment (stefansaraev) [inline xbmc/settings/Settings.cpp]: for the sake of consistency: bluray.playerregion or CSettings::SETTING_BLURAY_REGION :)

**Comment by ace20022** [comment]
@xhaggi @stefansaraev thanks!

**Comment by xhaggi** [comment]
np :smile:

**Comment by ace20022** [comment]
jenkins build this please

**Comment by ace20022** [comment]
Is anyone interested in bumping libbluray on the other platforms? It won't compile otherwise because of new defines.

**Comment by FernetMenta** [comment]
I am traveling with a MacBook and could try to do OSX tomorrow if @Memphiz is not faster.

**Comment by MartijnKaijser** [comment]
i've already uploaded the .tar to the mirror if that's ok
http://mirrors.kodi.tv/build-deps/sources/libbluray-0.8.1.tar.bz2

Need to adjust this makefile?https://github.com/xbmc/xbmc/blob/master/tools/depends/target/libbluray/Makefile

**Comment by ace20022** [comment]
@FernetMenta @MartijnKaijser thanks much! I'll adapt the makefile tomorrow.

**Comment by Memphiz** [comment]
@FernetMenta definitly not faster These days - appreciate if you Beat me to it ;)

**Comment by FernetMenta** [comment]
configure: error: BD-J requires ANT, but ant was not found. Please install it.

not sure how to handle this

**Comment by ace20022** [comment]
I built win32 without bdj. Config switches --disable-bdjava --without-libxml2 --without-freetype . I forgot that the options got inverted, I'll have to add these to the makefile.

**Comment by FernetMenta** [comment]
with this it compiles here on OSX64 https://github.com/FernetMenta/xbmc/commit/17af96f6c02c83fe5dab5c01e5d015eebe6e59c8

Had no chance for runtime testing.

**Comment by xhaggi** [comment]
stupid question, does it mean that we do not support bdj with that version or do we, but only if my local libbluray is compiled with bdj?

**Comment by ace20022** [comment]
We have never supported it, the three options were off by default in 0.2.1. But I hope the latter still works, I'll check that.

**Comment by ace20022** [comment]
@koying could you please have a look? setmntent is unknown on android: http://jenkins.kodi.tv/job/Android-ARM/5235/consoleFull

**Comment by koying** [comment]
I had the updated libbluray+java patch included a while ago.
Here is the android patch I used:
```
--- src/libbluray/bluray.c	2014-11-17 12:22:31.470370819 +0100
+++ src/libbluray/bluray.c	2014-11-17 13:33:16.093008686 +0100
@@ -1357,6 +1357,7 @@
     if (S_ISDIR(st.st_mode))
         return;
 
+#ifndef ANDROID
     FILE *f = setmntent ("/proc/self/mounts", "r");
     if (!f)
         return;
@@ -1376,6 +1377,7 @@
         }
     }
     endmntent (f);
+#endif
 }
 #endif
 #ifdef __APPLE__
@@ -2965,6 +2967,7 @@
 
 static int _play_bdj(BLURAY *bd, unsigned title)
 {
+#ifdef USING_BDJAVA
     int result;
 
     bd->title_type = title_bdj;
@@ -2978,6 +2981,12 @@
     }
 
     return result;
+#else
+    BD_DEBUG(DBG_BLURAY | DBG_CRIT, "Can't play BD-J title %d\n", title);
+    bd->title_type = title_undef;
+    _queue_event(bd, BD_EVENT_ERROR, BD_ERROR_BDJ);
+    return -1;
+#endif
 }
 
 static int _play_hdmv(BLURAY *bd, unsigned id_ref)
```

Not sure if still valid, though...

**Comment by ace20022** [comment]
jenkins build this please

**Comment by ace20022** [comment]
jenkins build this please

**Comment by ace20022** [comment]
jenkins build this please

Do we need runtime tests?

**Comment by fritsch** [comment]
@ace20022 Should I link your branch to this user here: http://forum.kodi.tv/showthread.php?tid=235876 so that he can test it? some of your commits might be able to positively influence this, e.g. https://github.com/ace20022/xbmc/commit/70658974cd5434bec1b0ed5d0f988b1dd14741ac

**Comment by ace20022** [comment]
@fritsch I think it's time for some debug log messages so we see if the disc gets played with menus etc.
I'll reply to the thread.

**Comment by ace20022** [comment]
Only addons on rbpi failed...

**Comment by MartijnKaijser** [comment]
great job.
Guess now we can provide some build on the forum and get it in at start of September?

**Comment by ace20022** [comment]
Right, I forgot about the merge windows ;-)  Android and osx runtime tests would be great.

**Comment by MartijnKaijser** [comment]
I'll do some droid testing. Kicked builds here http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-PR-Manually/149/

**Comment by Memphiz** [comment]
Can't test - have no blueray rips ...

**Comment by ace20022** [comment]
http://www.avsforum.com/forum/139-display-calibration/948496-avs-hd-709-blu-ray-mp4-calibration.html
The hdmv version

**Comment by MartijnKaijser** [comment]
Will need https://github.com/xbmc/xbmc/pull/7868 to be merged first before selection dialog works again

**Comment by MartijnKaijser** [comment]
so far everything works on my nvida shield (arm)

**Comment by da-anda** [comment]
sorry for hijacking this PR, but as it's somewhat related I thought I ask here. Would there be any objections in adding experimental BD-J support? Like have it disabled by default but users could enable it via a expert setting? Ofc given that we manage to compile it for every platform with BD-J.

btw - thanks @ace20022 for working on bluray playback improvements - much appreciated

**Comment by ace20022** [comment]
@da-anda some more qork is needed for that step, but it's on my list.

jenkins build this please

**Comment by MartijnKaijser** [comment]
fyi,
run tested on android with several bluray ISO and no issues

**Comment by ace20022** [comment]
@MartijnKaijser Thanks very much!

**Pull request merged**